### PR TITLE
Feat/update todoapp goldens

### PR DESCRIPTION
**Pull request by DDavidPrime**
#### Overview

> Closes: #863 

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [x] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [x] Added unit tests for new code and verified existing tests work as expected

*Code quality*
~- [ ] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`~
~- [ ] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations~
~- [ ] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes~

### CODE REVIEW

**Comment by DDavidPrime** [comment]
Closing this PR in favour of #883

File: examples/todoapp/test/feature_statistics/view/statistics_golden_test.dart [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+    buildScenario(
+      scenario: 'statistics_success',
+      widget: statisticsFactory(
+        todosStats: Stubs.todoListStatistics,
+      ),
+    ),
+    buildScenario(
+      scenario: 'statistics_loading',
+      customPumpBeforeTest: animationCustomPump,
+      widget: statisticsFactory(),
```
Comment (StanevPrime) [inline examples/todoapp/test/feature_statistics/view/statistics_golden_test.dart:22]: Do we still need the `animationCustomPump` ? If not, please remove this global function.

File: examples/todoapp/test/feature_todo_details/view/todo_details_golden_test.dart [hunk 2]
```diff
@@ -1,51 +1,58 @@
...
+      scenario: 'todo_details_empty',
+      widget: todoDetailsFactory(todo: Result.success(Stubs.todoEmpty)),
+    ),
+    buildScenario(
+      scenario: 'todo_details_success',
+      widget: todoDetailsFactory(todo: Result.success(Stubs.todoCompleted)),
+    ),
+    buildScenario(
+      scenario: 'todo_details_bulk_action_loading',
+      customPumpBeforeTest: animationCustomPump,
```
Comment (StanevPrime) [inline examples/todoapp/test/feature_todo_details/view/todo_details_golden_test.dart:22]: Same question as above.

**Review by StanevPrime — CHANGES_REQUESTED** [review]

**Review by StanevPrime — APPROVED** [review]

File: examples/todoapp/test/feature_statistics/view/statistics_golden_test.dart (hunk 1 above)
Comment (DDavidPrime) [reply examples/todoapp/test/feature_statistics/view/statistics_golden_test.dart:21]: Removed! 👍🏼

File: examples/todoapp/test/feature_todo_details/view/todo_details_golden_test.dart (hunk 2 above)
Comment (DDavidPrime) [reply examples/todoapp/test/feature_todo_details/view/todo_details_golden_test.dart]: Done!

**Pull request merged**